### PR TITLE
Feat/console-karnel-handle-and-register

### PR DESCRIPTION
**Pull request by SonyPradana**

### CODE REVIEW

File: tests/Integrate/Commands/HelpCommandsTest.php [hunk 1]
```diff
@@ -5,36 +5,44 @@
...
+                'mode'      => 'full',
+                'class'     => HelpCommand::class,
+                'fn'        => 'main',
+            ],
+        ];
+
+        $helpCommand = new HelpCommand(['cli', '--help']);
         ob_start();
-        $exit = $helpCommand->main();
+        $exit = $helpCommand->{'main'}();
```
Comment (anggerpradana) [inline tests/Integrate/Commands/HelpCommandsTest.php:45]: 😅 but is oke

File: src/System/Integrate/Console/HelpCommand.php [hunk 2]
```diff
@@ -263,11 +261,9 @@ public function commandHelp(): int
...
-        $commandmap =  [];
-        foreach ($this->commands as $command) {
-            $commandmap[] = new CommandMap($command);
-        }
-
-        return $commandmap;
+        return array_map(
+            static fn ($command): CommandMap => new CommandMap($command),
+            Application::getIntance()[ConfigRepository::class]->get('commands', [])
+        );
```
Comment (anggerpradana) [inline src/System/Integrate/Console/HelpCommand.php:267]: intead using `Application::getIntance()` better using application parameter

File: src/System/Integrate/Console/Karnel.php [hunk 3]
```diff
@@ -102,6 +105,36 @@ public function bootstrap(): void
...
+    /**
+     * Call command using know signature.
+     * The signature doset require php as prefix.
+     * For better parse use `handle` method istead.
+     *
+     * @param array<string, string|bool|int|null> $parameter
+     *
+     * @since v0.33
+     */
+    public function call(string $signature, array $parameter = []): int
```
Comment (anggerpradana) [inline src/System/Integrate/Console/Karnel.php:117]: oke but how about `run` is more representative

**Review by anggerpradana — CHANGES_REQUESTED** [review]

File: src/System/Integrate/Console/HelpCommand.php (hunk 2 above)
Comment (SonyPradana) [reply src/System/Integrate/Console/HelpCommand.php:267]: for now it is oke, the problem is parent `Command::main()` doesn't required any parameters. if it change will throw error due child not same with parent class.

The solutions is make pr to remove `Command::main()` from Command. so child can use costume parameter.

or

dont use main methods to register this signature.

**Pull request merged**